src/datasets/arrow_dataset.py

```diff
@@ -2742,10 +2743,12 @@ def prepare_for_task(self, task: Union[str, TaskTemplate], id: int = 0) -> "Data
         dataset = dataset.cast(features=template.features)
         return dataset
 
-    def _getitem(self, key: Union[int, slice, str], **kwargs) -> Union[Dict, List]:
+    def _getitem(self, key: Union[int, slice, str, Collection[int]], **kwargs) -> Union[Dict, List]:
```

Comment:
I think Sequence makes more sense, we shouldn't let users think they can should use a dictionary here
```suggestion
    def _getitem(self, key: Union[int, slice, str, Sequence_[int]], **kwargs) -> Union[Dict, List]:
```

Reply:
@lhoestq, I agree with you. Note that I added `Sequence` in a previous commit: https://github.com/huggingface/datasets/pull/5859/commits/a4c43e8b33e2d25e2ee702bd8b358fc6e568ad55

However, then I realized that `Sequence` is too restrictive: we also support `np.ndarray` (and also `pd.Series`) and these are not `Sequence` (but they are `Collection`). See: https://github.com/huggingface/datasets/blob/22d1d533e8ab831b1aa1aab3e7d3c72ba42a83e8/tests/test_arrow_dataset.py#L212

That is why I then changed it to `Collection`. 
But I agree that `Collection` is too permissive (we do not support `dict`).

What do you think could be the better solution? Maybe, instead of `Collection[int]`:
```python
ListLike[int]
```
See: https://github.com/pandas-dev/pandas/blob/2a270d8b04d9573b17b9d550a049d003a974a9d4/pandas/_typing.py#LL114C1-L114C9

Reply:
ListLike sounds good !